management/virtualmachine/client.go

```diff
@@ -118,6 +119,25 @@ func (vm VirtualMachineClient) GetRole(cloudServiceName, deploymentName, roleNam
 	return role, nil
 }
 
+// AddRole adds a Virtual Machine to a deployment of Virtual Machines, where role name = VM name
+// See https://msdn.microsoft.com/en-us/library/azure/jj157186.aspx
+func (vm VirtualMachineClient) AddRole(cloudServiceName string, deploymentName string, role Role) (management.OperationID, error) {
+	if cloudServiceName == "" {
```

Comment:
Thanks @ahmetalpbalkan - This method is almost a mirror copy of the existing "UpdateRole" - which takes a Role parameter, then serializes a PersistentVMRole instance made out of that Role parameter. IMHO, we should stick to the same data types for consistency. Please let me know what you feel.